From: Sjoerd Simons <sjoerd.simons@collabora.co.uk>
Date: Thu, 25 Jun 2015 10:39:48 +0200
Subject: [PATCH] Rely on gdk to dispatch wayland events

Disable cogl dispatching wayland events as Gdk will already be
dispatching them causing a deadlock when cogl tries again to dispatch
wayland events.

https://bugzilla.gnome.org/show_bug.cgi?id=774079
---
 clutter/gdk/clutter-backend-gdk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clutter/gdk/clutter-backend-gdk.c b/clutter/gdk/clutter-backend-gdk.c
index 25c687b..f81b85b 100644
--- a/clutter/gdk/clutter-backend-gdk.c
+++ b/clutter/gdk/clutter-backend-gdk.c
@@ -275,7 +275,7 @@ clutter_backend_gdk_get_renderer (ClutterBackend  *backend,
       /* Force a Wayland winsys */
       cogl_renderer_set_winsys_id (renderer, COGL_WINSYS_ID_EGL_WAYLAND);
       cogl_wayland_renderer_set_foreign_display (renderer, display);
-      cogl_wayland_renderer_set_event_dispatch_enabled (renderer, !disable_event_retrieval);
+      cogl_wayland_renderer_set_event_dispatch_enabled (renderer, FALSE);
     }
   else
 #endif
